From 9a11bda77217a7bdb55b8bd736969355edb7efdd Mon Sep 17 00:00:00 2001
From: Chih-Hung Hsieh <chh@google.com>
Date: Tue, 2 Aug 2022 12:03:50 -0700
Subject: [PATCH 06/11] Update clang version to clang-r468909

* Allow/show clang deprecated* and array-parameter warnings.

Upstream commit date: 2022-07-25

Test: presubmit
Bug: 241941550
Bug: 241601211
Bug: 241011085
Change-Id: I5c152823500dde656e1306afcb6f45e824b50102
---
 cc/config/global.go | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/cc/config/global.go b/cc/config/global.go
index a81d466c5..340ec21c0 100644
--- a/cc/config/global.go
+++ b/cc/config/global.go
@@ -243,6 +243,10 @@ var (
 		"-Wno-error=unused-but-set-parameter", // http://b/197240255
 		// New warnings to be fixed after clang-r458507
 		"-Wno-error=unqualified-std-cast-call", // http://b/239662094
+		// New warnings to be fixed after clang-r468909
+		"-Wno-error=array-parameter",     // http://b/241941550
+		"-Wno-error=deprecated-builtins", // http://b/241601211
+		"-Wno-error=deprecated",          // in external/googletest/googletest
 	}
 
 	noOverrideExternalGlobalCflags = []string{
@@ -305,8 +309,8 @@ var (
 
 	// prebuilts/clang default settings.
 	ClangDefaultBase         = "prebuilts/clang/host"
-	ClangDefaultVersion      = "clang-r458507"
-	ClangDefaultShortVersion = "15.0.1"
+	ClangDefaultVersion      = "clang-r468909"
+	ClangDefaultShortVersion = "15.0.2"
 
 	// Directories with warnings from Android.bp files.
 	WarningAllowedProjects = []string{
-- 
2.38.1

